kubespawner/utils.py

```diff
@@ -1,132 +1,28 @@
 """
 Misc. general utility functions, not tied to Kubespawner directly
 """
-import os
-import yaml
+import random
+import hashlib
 
-from tornado.httpclient import HTTPRequest
-
-
-def request_maker():
-    """
-    Return a k8s api aware HTTPRequest factory that autodiscovers connection info
-    """
-    if os.path.exists('/var/run/secrets/kubernetes.io/serviceaccount/token'):
-        # We are running in a pod, and have access to a service account!
-        return request_maker_serviceaccount()
-    else:
-        return request_maker_kubeconfig()
-
-
-def request_maker_serviceaccount():
+def generate_hashed_slug(slug, limit=63, hash_length=6):
     """
-    Return a k8s api aware HTTPRequest factory that discovers connection info from a service account
+    Generate a unique name that's within a certain length limit
 
-    Discovers the hostname, port, protocol & authentication details for talking to
-    the kubernetes API from a ServiceAccount. This requires that service accounts are
-    turned on in your kubernetes cluster and that the calling code is running in a pod.
+    Most k8s objects have a 63 char name limit. We wanna be able to compress
+    larger names down to that if required, while still maintaining some
+    amount of legibility about what the objects really are.
     """
-    with open('/var/run/secrets/kubernetes.io/serviceaccount/token') as f:
-        token = f.read()
-    api_url = 'https://{host}:{port}'.format(
-        host=os.environ['KUBERNETES_SERVICE_HOST'],
-        port=os.environ['KUBERNETES_SERVICE_PORT']
-    )
+    # if the length of the slug is shorter than limit - hash_length, we just
+    # use it directly. Good enough!
+    if len(slug) < (limit - hash_length):
+        return slug
 
-    def make_request(url, **kwargs):
-        """
-        Make & return a HTTPRequest object suited to making requests to a Kubernetes cluster
+    # If not, we pick the first limit - hash_length chars from slug & hash the rest.
```

Comment:
Put what approach we are using in the docstring if > 63 char

Reply:
Done.

Reply:
Done